src/com/shmuelzon/HomeAssistantFloorPlan/Controller.java

```diff
@@ -949,8 +949,12 @@ private List<Set<Entity>> findIntersectingStateIcons() {
 
     private Point2d getCenterOfStateIcons(Set<Entity> entities) {
         Point2d centerPosition = new Point2d();
-        for (Entity entity : entities )
+
+        for (Entity entity : entities ) {
+            // Round position-values because the entity's stored position can differ fractionally from the position in SH3D.
+            entity.position.set(Math.round(entity.position.x), Math.round(entity.position.y));
             centerPosition.add(entity.position);
+        }
```

Comment:
I'm not sure why this is a problem or even considered a bug.
If two objects were really in the same position (in all 3 axes) and had the same height their 2D projected location should also be calculated the same. IIRC, there's no random elements here.

Reply:
I don't know why there's a difference between the position in SH3D and the script either.
Perhaps due to converting units (cm to coordinates) in SH3D?

In SH3D I have 3 pieces of furniture with the following position (x, y) and dimensions (w, d, h): 350cm, 42.7cm, 200cm / 8.5cm, 1.1cm, 8.5cm.

In the script they have the following positions (Point2D):
1. (954.3064556411717, 712.7039570614329) < Differs
2. (954.3475272573851, 712.6856725617578) < same
3. (954.3475272573851, 712.6856725617578) < same

In `home.xml` they are:
1. x='350.0056' y='42.678185'  < Differs
2. x='349.97675' y='42.691845' < Same
3. x='349.97675' y='42.691845' < Same

The only way that came up to me to make the same again as they are in SH3D, was to round the values. Rounding them would not change the position significantly and won't be noticed in the results.

Reply:
OK, so if they differ in `Home.xml` it means they're actually different. Maybe SH3D's UI rounds them so they appear the same value in the UI but underneath, they're different. For this plugin, I don't think it really matters if it's rounded or not since it's actually better for us if they're slightly different and, by the time it's converted to the top/left percentages, it probably doesn't really matter.